rust/libnewsboat-ffi/src/lib.rs

```diff
@@ -146,6 +146,27 @@ pub extern "C" fn rs_trim(input: *const c_char) -> *mut c_char {
     result.into_raw()
 }
 
+#[no_mangle]
+pub extern "C" fn rs_quote(input: *const c_char) -> *mut c_char {
+
+    let rs_input = unsafe { CStr::from_ptr(input) };
+    let rs_input = rs_input.to_string_lossy().into_owned();
+
+    let output = utils::quote(rs_input);
+    let output = CString::new(output).unwrap();
```

Comment:
`unwrap()` will panic if the value is `None`, so we avoid it; if we can't (as is the case here), we write a comment to explain why we're sure it won't panic. 

In this case, `output` is the same as `rs_input` plus a pair of quotes. `CString::new` will return `None` if there are null bytes in the string. Since `rs_input` we got from C++ (so no null bytes inside), and we only added a pair of quotes, we can be sure this won't panic.

Reply:
Copy-paste is fine, but please update the variable names and mention that `output` also contains backslashes and quotes (which obviously aren't null bytes).

Reply:
Whoops! That was entirely me apparently paying no attention.